lua/nvim-tree/explorer/live-filter.lua

```diff
@@ -2,17 +2,27 @@ local view = require "nvim-tree.view"
 local utils = require "nvim-tree.utils"
 local Iterator = require "nvim-tree.iterators.node-iterator"
 
-local M = {
-  filter = nil,
-}
+local LiveFilter = {}
```

Comment:
Could you please add luadoc here?

We can then put `---@field live_filter LiveFilter` into `@class Explorer`

That will allow autocompletion and (more importantly) language server validations/checks.

Reply:
I was wondering for what they are used for, thanks for clarification!